### PR TITLE
chore: replace `reorder-python-imports` with `isort` to address `black` incompatibility

### PR DESCRIPTION
**Pull request by NiceAesth**
# Describe your changes

Can unlock black version with this

## Checklist
- [x] I've manually tested my code

### CODE REVIEW

**Comment by NiceAesth** [comment]
We can disable adding the `__future__ import annotations` import now via comments

**Comment by cmyui** [comment]
What's your motivation to switch to `isort`? My choice of `reorder-python-imports` was in the vein of minimizing merge conflicts. Also tool quality i guess -- i highly regard/trust asottile haha

**Comment by minisbett** [comment]
> What's your motivation to switch to `isort`? My choice of `reorder-python-imports` was in the vein of minimizing merge conflicts. Also tool quality i guess -- i highly regard/trust asottile haha

Honestly I've just googled for "python linq equivalent" and found this XD

**Comment by NiceAesth** [comment]
The fact it completely breaks black and neither maintainer wants to support the other tool out of stubbornness

This also does pretty much the same import style anyway just with a couple behaviour changes

**Comment by NiceAesth** [comment]
Forgot to link to this issue in the original post

https://github.com/psf/black/issues/4175

File: app/api/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # type: ignore
-from __future__ import annotations
+# isort: dont-add-imports
```
Comment (cmyui) [inline app/api/__init__.py:2]: Know if there's a way to automagically handle `__init__.py` files globally in the `.isort.cfg`?

**Comment by cmyui** [comment]
> The fact it completely breaks black and neither maintainer wants to support the other tool out of stubbornness
> 
> This also does pretty much the same import style anyway just with a couple behaviour changes
> 
> * Will sort imports in stuff like `TYPE_CHECKING` blocks
> * Has support for disabling certain features per-file via comments e.g. disabling adding imports (i think reorder-python-imports also had this but on a simpler level)
> * Has configs for sort behaviour (so we can use it with black)
> 
> the other alternatives are switching to ruff or forking the tool to make it compatible and i view neither of these as necessary personally. isort seemed to be the lowest lift solution for resolving the incompatibility

Interesting okay -- damn that's unfortunate indeed, but understood. I see this PR aligns with choices made by many other familiar faces in the py ecosystem, so I'm aligned with the change.

+1 👍🏻

File: app/api/__init__.py (hunk 1 above)
Comment (NiceAesth) [reply app/api/__init__.py:2]: seems like there is no way to do this sadly

**Review by tsunyoku — APPROVED** [review]

**Pull request merged**